lib/core/connector_job.rb

```diff
@@ -111,6 +118,33 @@ def reload
       @connector = nil
     end
 
+    def with_lock
+      seq_no = nil
+      primary_term = nil
+
+      response = ElasticConnectorActions.get_job(id)
+
+      yield response, seq_no, primary_term
+    end
```

Comment:
So it's not a real lock, it's optimistic concurrency, right? Method name is a bit confusing

Reply:
Yes, you're right. 

I must say the method name is misleading :). 

would `with_concurrency_control` work? any ideas?